pandas/core/window/rolling.py

```diff
@@ -473,12 +504,31 @@ def _apply(
 
                 def calc(x):
                     x = np.concatenate((x, additional_nans))
-                    min_periods = calculate_min_periods(
-                        window, self.min_periods, len(x), require_min_periods, floor
+                    if not isinstance(window, BaseIndexer):
+                        min_periods = calculate_min_periods(
+                            window, self.min_periods, len(x), require_min_periods, floor
+                        )
+                    else:
+                        min_periods = calculate_min_periods(
+                            self.min_periods or 1,
+                            self.min_periods,
+                            len(x),
+                            require_min_periods,
+                            floor,
+                        )
+                    start, end = window_indexer.get_window_bounds(
+                        num_values=len(x),
+                        min_periods=self.min_periods,
+                        center=self.center,
+                        closed=self.closed,
                     )
-                    start, end = window_indexer(
-                        x, window, self.closed, index_as_array
-                    ).get_window_bounds()
+                    if np.any(np.diff(start) < 0) or np.any(np.diff(end) < 0):
```

Comment:
you can just set is_monotonic_bounds=np.any(.......)

and pass that in line 482

Reply:
Will need to add `is_monotonic_bounds` as an argument to all the "fixed" functions in Cython, but I suppose that makes this cleaner.

Reply:
Actually I cannot easily modify this yet without another refactor. `func` here isn't always a cython function that could accept `is_monotonic_bounds`; sometimes it's a python wrapped function like `apply_func` or `_zqrt`. I can look into this in another refactor.

Reply:
k, followup is ok